include/simulation_time/simulation_clock.hpp

```diff
@@ -0,0 +1,219 @@
+#pragma once
+
+#include <cassert>
+#include <chrono>
+#include <stdexcept>
+#include <utility>
+
+namespace ngen {
+
+//! Simulation Clock
+//!
+//! Represents simulation ticks monotonically increasing from zero, and
+//! provides conversion utilities between simulation ticks and
+//! calendar time.
+//!
+//! Implements named requirements: Clock (https://en.cppreference.com/w/cpp/named_req/Clock)
+//!
+struct simulation_clock final
+{
+    /* Clock named requirements */
+
+    using rep = std::int64_t; // TODO: double vs std::int64_t
+    using period = std::ratio<1>;
+    using duration = std::chrono::duration<rep, period>;
+    using time_point = std::chrono::time_point<simulation_clock, duration>;
+    static constexpr bool is_steady = false;
+
+    //! Current Simulation Time
+    //! \returns The current point in time the simulation is at.
```

Comment:
This is almost philosophical, but in a sense, the description of the returned value is premised on the expected use by callers of `now()` and `tick()` and the epoch routines.

To be really strict, it's really more like "The accumulation of argument values from preceding calls to `tick()`"